lib/pf/factory/scan.pm

```diff
@@ -0,0 +1,80 @@
+package pf::factory::scan;
+
+=head1 NAME
+
+pf::factory::scan
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::factory::scan
+
+The factory for creating pf::scan objects
+
+=cut
+
+use strict;
+use warnings;
+use Module::Pluggable search_path => 'pf::scan', sub_name => 'modules' , require => 1, except => qr/^pf::scan::wmi::(.*)$/;
+use List::MoreUtils qw(any);
+use pf::scan;
+use pf::ConfigStore::Scan;
+
+our @MODULES = __PACKAGE__->modules;
+
+sub factory_for { 'pf::scan' }
+
+sub configStoreClass { 'pf::ConfigStore::Scan' }
+
+sub new {
+    my ($class,$name) = @_;
+    my $object;
+    my $configStore = $class->configStoreClass->new;
```

Comment:
Yes I know but we now need to use the hashes instead. See #422 in lib/pf/factory for the corrected version